src/sentry/notifications/helpers.py

```diff
@@ -610,3 +641,149 @@ def get_providers_for_recipient(
     return user_providers
+
+
+def recipient_is_user(recipient: RpcActor | Team | User) -> bool:
+    from sentry.interfaces.user import User
+
+    if isinstance(recipient, RpcActor) and recipient.actor_type == ActorType.USER:
+        return True
+    return isinstance(recipient, (RpcUser, User))
+
+
+def recipient_is_team(recipient: RpcActor | Team | User) -> bool:
+    from sentry.models.team import Team
+
+    if isinstance(recipient, RpcActor) and recipient.actor_type == ActorType.TEAM:
+        return True
+    return isinstance(recipient, Team)
+
+
+def get_all_setting_providers(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> Iterable[NotificationSettingProvider]:
+    project_settings = (
+        Q(scope_type=NotificationScopeType.PROJECT.value, scope_identifier=project.id)
+        if project
+        else Q()
+    )
+
+    org_settings = (
+        Q(
+            scope_type=NotificationScopeType.ORGANIZATION.value,
+            scope_identifier=project.organization.id,
+        )
+        if organization
+        else Q()
+    )
+
+    team_or_user_settings = Q()
+    if recipient_is_user(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.USER.value, scope_identifier=recipient.id
+        )
+    elif recipient_is_team(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.TEAM.value, scope_identifier=recipient.id
+        )
+
+    return NotificationSettingProvider.objects.filter(
+        project_settings | org_settings | team_or_user_settings
+    )
+
+
+def get_all_setting_options(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+):
+    project_settings = (
+        Q(scope_type=NotificationScopeType.PROJECT.value, scope_identifier=project.id)
+        if project
+        else Q()
+    )
+
+    org_settings = (
+        Q(
+            scope_type=NotificationScopeType.ORGANIZATION.value,
+            scope_identifier=project.organization.id,
+        )
+        if organization
+        else Q()
+    )
+
+    team_or_user_settings = Q()
+    if recipient_is_user(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.USER.value, scope_identifier=recipient.id
+        )
+    elif recipient_is_team(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.TEAM.value, scope_identifier=recipient.id
+        )
+
+    NotificationSettingOption.objects.filter(
+        project_settings | org_settings | team_or_user_settings
+    )
+
+
+def get_settings_for_recipient(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> MutableMapping[NotificationScopeType, NotificationSettingOptionValues]:
```

Comment:
@snigdhas Can we use the new enums here? NotificationScopeEnum and NotificationSettingsOptionEnum instead of the old ones which map to numbers.